internal/state/graph/reference_grant.go

```diff
@@ -0,0 +1,64 @@
+package graph
+
+import (
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/gateway-api/apis/v1beta1"
+)
+
+func refGrantAllowsGatewayToSecret(
```

Comment:
do we need to think about a more efficient representation for reference grants, so that we have O(1) check instead of O(N) ?

Reply:
Probably, but I think this can wait until the next ReferenceGrant issue so I have a better chance of getting the structure right.